src/topp/IsobaricWorkflow.cpp

```diff
@@ -0,0 +1,662 @@
+      for (Size i = 0; i < qc.size(); ++i)
+      {
+        qc[i].mz_deltas = std::vector<double>(pep_ids.size());
+      }
+
+      cur_cmap.resize(pep_ids.size());
+
+      channel_extractor.registerChannelsInOutputMap(cmap, mz_file);
+      // add filename references
+      for (auto& column : cur_cmap.getColumnHeaders())
+      {
+        column.second.filename = mz_file;
+      }
+
+      #ifdef _OPENMP
+      #pragma omp parallel for /*num_threads(inner_threads)*/
+      #endif
+      for (Size pep_idx = 0; pep_idx < pep_ids.size(); ++pep_idx)
+      {
+        auto& pep = pep_ids[pep_idx];
+        const auto& spec_ref = pep.getSpectrumReference();
+        if (!spec_ref.empty())
+        {
+          auto ms2spec_it = ms2scan_to_index.find(spec_ref);
+          if (ms2spec_it != ms2scan_to_index.end())
+          {
+            std::vector<std::pair<double,unsigned>> channel_qc(quant_method->getNumberOfChannels(), std::make_pair(std::numeric_limits<double>::quiet_NaN(), 0));
+
+            auto [quant_spec_idx, id_spec_idx, ms1_spec_idx] = getSpecIdxs_(ms2spec_it->second, exp, has_ms3);
+            auto [quant_purity, id_purity] = getPurities_(quant_spec_idx, id_spec_idx, ms1_spec_idx, exp, has_ms3, max_precursor_isotope_deviation, calc_id_purity, interpolate_precursor_purity);
+
+            if (has_ms3 && exp[quant_spec_idx].getMSLevel() != 3)
+            {
+              throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "MS3 spectrum expected but not found.", String(exp[quant_spec_idx].getMSLevel()));
+            }
+
+            std::vector<double> itys = channel_extractor.extractSingleSpec(quant_spec_idx, exp, channel_qc);
+
+            // TODO if itys are all zero we can actually skip correction and quantification
+            auto m = correction_matrix.getEigenMatrix();
+            std::vector<double> corrected(itys.size(), 0.);
+            NonNegativeLeastSquaresSolver::solve(m, itys, corrected);
+            fillConsensusFeature_(cur_cmap[pep_idx], pep, exp, id_spec_idx, quant_spec_idx, corrected, quant_method, quant_purity, id_purity, min_reporter_intensity, i);
+            for (Size i = 0; i < channel_qc.size(); ++i)
+            {
+              // TODO ProteomeDiscoverer also outputs:
+              //  - Reporter S/N but with S/N they mean the intensity corrected for the noise value from the raw thermo Orbitrap files
+              //    (and I dont think we read them, see https://github.com/compomics/ThermoRawFileParser/blob/c293d4aa1b04bfd62124ff42c512572427a4316a/Writer/MzMlSpectrumWriter.cs#L1664)
+              //  - For SPS-MS3 the number of precursor windows that actually surround a fragment from the identified peptide! Useful, but currently not implemented here.
+              qc[i].mz_deltas[pep_idx] = channel_qc[i].first;
+              if (channel_qc[i].second > 1)
+              {
+                #ifdef _OPENMP
+                #pragma omp atomic
+                #endif
```

Comment:
```suggestion
                #pragma omp atomic
```

no need for ifdefs for pragma; ifdefs for omp functions are a sensible though